homeassistant/components/romy/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Config flow for ROMY integration."""
+from __future__ import annotations
+
+import romy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, LOGGER
+
+
+def _schema_with_defaults(
+    host: str = "", port: int = 8080, name: str = ""
+) -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST, default=host): cv.string,
+            vol.Optional(CONF_NAME, default=name): cv.string,
+        },
+    )
+
+
+def _schema_with_defaults_and_password(
+    host: str = "", port: int = 8080, name: str = "", password: str = ""
+) -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST, default=host): cv.string,
+            vol.Optional(CONF_NAME, default=name): cv.string,
+            vol.Required(CONF_PASSWORD, default=password): vol.All(str, vol.Length(8)),
+        },
+    )
+
+
+class RomyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for ROMY."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Handle a config flow for ROMY."""
+        self.discovery_schema = None
+        self.host: str = ""
+        self.name: str = ""
+        self.password: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+        data = self.discovery_schema or _schema_with_defaults()
+
+        if user_input is not None:
+            if not errors:
+                ## Save the user input and finish the setup
+                self.host = user_input["host"]
+                self.name = user_input["name"]
+                if "password" in user_input:
+                    self.password = user_input["password"]
+
+                new_romy = await romy.create_romy(self.host, self.password)
+
+                if not new_romy.is_initialized:
+                    errors[CONF_HOST] = "cannot_connect"
+                    return self.async_show_form(
+                        step_id="user", data_schema=data, errors=errors
+                    )
+
+                if not new_romy.is_unlocked:
+                    errors[CONF_PASSWORD] = "invalid_auth"
+                    return self.async_show_form(
+                        step_id="user", data_schema=data, errors=errors
+                    )
+
+                return self.async_create_entry(
+                    title=user_input["name"], data=user_input
+                )
```

Comment:
```suggestion
                if not errors:
	            return self.async_create_entry(
	                title=user_input["name"], data=user_input
	            )
```

To simplify the code